src/core/reactor.cc

```diff
@@ -1904,125 +1904,96 @@ reactor::make_pipe() {
+                               [](auto& s) { return s.c_str(); });
+                envp.push_back(nullptr);
+
+                int r = 0;
+                r = ::posix_spawn_file_actions_init(&actions);
+                throw_pthread_error(r);
+                // the child process does not write to stdin
+                std::get<pipefd_write_end>(cin_pipe).spawn_action_add_close(&actions);
+                // the child process does not read from stdout
+                std::get<pipefd_read_end>(cout_pipe).spawn_action_add_close(&actions);
+                // the child process does not read from stderr
+                std::get<pipefd_read_end>(cerr_pipe).spawn_action_add_close(&actions);
+                // redirect stdin, stdout and stderr to cin_pipe, cout_pipe and cerr_pipe respectively
+                std::get<pipefd_read_end>(cin_pipe).spawn_action_add_dup2(&actions, STDIN_FILENO);
+                std::get<pipefd_write_end>(cout_pipe).spawn_action_add_dup2(&actions, STDOUT_FILENO);
+                std::get<pipefd_write_end>(cerr_pipe).spawn_action_add_dup2(&actions, STDERR_FILENO);
+                // after dup2() the interesting ends of pipes, close them
+                std::get<pipefd_read_end>(cin_pipe).spawn_action_add_close(&actions);
+                std::get<pipefd_write_end>(cout_pipe).spawn_action_add_close(&actions);
+                std::get<pipefd_write_end>(cerr_pipe).spawn_action_add_close(&actions);
+
+                r = ::posix_spawnattr_init(&attr);
+                throw_pthread_error(r);
+                // make sure the following signals are not ignored by the child process
+                sigset_t default_signals;
+                sigemptyset(&default_signals);
+                sigaddset(&default_signals, SIGINT);
+                sigaddset(&default_signals, SIGTERM);
+                r = ::posix_spawnattr_setsigdefault(&attr, &default_signals);
+                throw_pthread_error(r);
+                // make sure no signals are marked in the child process
+                sigset_t mask_signals;
+                sigemptyset(&mask_signals);
+                r = ::posix_spawnattr_setsigmask(&attr, &mask_signals);
+                throw_pthread_error(r);
+                r = ::posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGDEF | POSIX_SPAWN_SETSIGMASK);
+                throw_pthread_error(r);
+
+                return _thread_pool->submit<syscall_result<int>>([&child_pid, &pathname, &actions, &attr,
+                                                                  argv = std::move(argvp),
+                                                                  env =  std::move(envp)] {
+                    return wrap_syscall<int>(::posix_spawn(&child_pid, pathname.c_str(), &actions, &attr,
+                                                           const_cast<char* const *>(argv.data()),
+                                                           const_cast<char* const *>(env.data())));
```

Comment:
This hunk is impossible to review, because it changes things and indents them in on go. Please, split -- first patch adds/removes essential pieces, but leaves the indentation broken in order to minimize the '-/+' lines. Second patch fixes the indentation.

Reply:
@xemul thanks for your review. and sorry that i forgot this. should have split the commit first as you pointed this out before.